testApp/src/App.svelte

```diff
@@ -0,0 +1,122 @@
+<!-- App.svelte -->
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import Gallery from './Components/Gallery.svelte';
+	import MovieDetail from './Components/MovieDetail.svelte';
+	import NavBar from './Components/NavBar.svelte';
+	import '/global.css';
+	import type { Movie } from '/Interfaces/Movie';
+	import type { Franchise } from '/Interfaces/Franchise';
+
+	export let bearer: string;
+
+	let movies: Movie[] = [];
+	let currentIndex: number = -1;
+	let showDetail: boolean = false;
+	let currentMovie: Movie = null;
```

Comment:
I agree this is a bit sloppy I changed it to movie = movies[currentindex].

https://github.com/BowenGeense/MovieShowcase/pull/3/files#diff-1e1270da740e81dd13f8a65057582b0fce8a5fb3817ef913f3e90d7f82c4e9df

Reply:
`movies[-1]` can also be undefined, because there is an empty array from the get-go. Maybe it should be typed as `Movie | undefined`.

Reply:
Thank you for the tip it would indeed only make sense for it to be "Movie | undefined"
i implemented this : 

https://github.com/BowenGeense/MovieShowcase/pull/3/files#diff-1e1270da740e81dd13f8a65057582b0fce8a5fb3817ef913f3e90d7f82c4e9df